### PR TITLE
DHFPROD-2806: Added json-mapping-example project

### PR DESCRIPTION
**Pull request by rjrudin**
Also updated the insurance project to use the new mapping system

### CODE REVIEW

**Comment by bsrikan** [comment]
I get following error when running the jsonToJson flow. Followed the readme. Ran hubinit, mlDeploy then login to QS to run flow:

`{"stack":"Error: JS-JAVASCRIPT: /*:stylesheet/*:template[4] <xsl:for-each select=\"/\"> <xsl:call-template name=\"Person\"/></xsl:for-each> -- Error running JavaScript request: Error: Dictionary not found at '/referenceData/countryCodes.json'\n at main (/data-hub/5/builtins/steps/mapping/entity-services/main.sjs:72:11)\n at Flow.runMain (/data-hub/5/impl/flow.sjs:458:12)\n at Flow.runStep (/data-hub/5/impl/flow.sjs:341:58)\n at Flow.runFlow (/data-hub/5/impl/flow.sjs:210:12)\n at post (/marklogic.rest.resource/mlRunFlow/assets/resource.sjs:60:25)\n at callExtension (/MarkLogic/rest-api/lib/extensions-util.sjs:93:30)\n at applyOnce (/MarkLogic/rest-api/lib/extensions-util.sjs:111:10)","message":"JS-JAVASCRIPT: /*:stylesheet/*:template[4] <xsl:for-each select=\"/\"> <xsl:call-template name=\"Person\"/></xsl:for-each> -- Error running JavaScript request: Error: Dictionary not found at '/referenceData/countryCodes.json'","name":"Error"} `

File: examples/json-mapping-example/build.gradle [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    repositories {
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        jcenter()
+    }
+
+    dependencies {
+        classpath "net.saliman:gradle-properties-plugin:1.5.1"
+        if (project.hasProperty("testing")) {
+            classpath "com.marklogic:ml-data-hub:5.2-SNAPSHOT"
```
Comment (rjrudin) [inline examples/json-mapping-example/build.gradle]: Bumped this up to 5.2-SNAPSHOT

**Comment by rjrudin** [comment]
@bsrikan I bumped up the version to 5.2-SNAPSHOT.

Also, note the instructions in the README - you have to use Gradle to install this. It depends on loading some reference data under src/main/ml-data. QS won't load that correctly unless you run QS from the project directory.

**Comment by bsrikan** [comment]
> @bsrikan I bumped up the version to 5.2-SNAPSHOT.
> 
> Also, note the instructions in the README - you have to use Gradle to install this. It depends on loading some reference data under src/main/ml-data. QS won't load that correctly unless you run QS from the project directory.

I did use gradle to install and then log in via QS to run flow. Will try again. @ryanjdew @srinathgit did you guys run into this issue as well or is it just me?

**Comment by rjrudin** [comment]
> I did use gradle to install and then log in via QS to run flow. Will try again. @ryanjdew @srinathgit did you guys run into this issue as well or is it just me?

Oh wait - I know the problem, I had to add some properties to gradle.properties, but that file is gitignored. Will fix.

**Comment by rjrudin** [comment]
@wooldridge  I added the fix for 3607 to this.

**Pull request merged**